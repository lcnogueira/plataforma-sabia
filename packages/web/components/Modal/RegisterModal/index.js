import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useRouter } from 'next/router';
import { useTranslation } from 'react-i18next';
import { MdPermContactCalendar, MdMailOutline, MdVpnKey } from 'react-icons/md';
import { AiFillCloseCircle as CloseIcon } from 'react-icons/ai';

import Link from 'next/link';
import { toast } from '../../Toast';
import { Form, InputField, CheckBoxField } from '../../Form';
import { Button } from '../../Button';
import { SafeHtml } from '../../SafeHtml';
import {
	StyledRegisterModal,
	StyledLabel,
	StyledCloseButton,
	StyledModalContent,
	ActionsRegister,
	Label,
	LabelGroups,
	StyledSpan,
	StyledLink,
	ErrorMessage,
	SuccessMessage,
} from './styles';
import { useModal, useAuth } from '../../../hooks';

const RegisterModal = ({ closeModal }) => {
	const { openModal } = useModal();
	const { register } = useAuth();
	const [loading, setLoading] = useState(false);
	const { t } = useTranslation(['common']);
	const [message, setMessage] = useState('');
	const [error, setError] = useState(false);

	const router = useRouter();

	const handleSubmit = async ({ fullname, email, password }) => {
		setLoading(true);
		const result = await register({ fullname, email, password });
		setLoading(false);
		setError(false);
		if (result.error) {
			setError(true);
			setMessage(result.error.message[0].message);
		} else {
			closeModal();
			toast.success(t('common:accountCreated'));
			router.push('/auth/confirm-account');
		}
	};

	const [acceptedTerms, setAcceptedTerms] = useState({
		terms_conditions: false,
		data_conditions: false,
		platform_conditions: false,
		services_conditions: false,
		channel_conditions: false,
		link_conditions: false,
		process_conditions: false,
	});

	// eslint-disable-next-line consistent-return
	const handleAcceptedTerms = (type) => {
		const types = Object.keys(acceptedTerms);

		if (!type || !types.includes(type)) return;

		setAcceptedTerms({
			...acceptedTerms,
			[type]: !acceptedTerms[type],
		});
	};

	const isMissingCheck = () => {
		const checkTerms = [
			'terms_conditions',
			'data_conditions',
			'platform_conditions',
			'services_conditions',
			'channel_conditions',
			'link_conditions',
		];
		return checkTerms.every((key) => acceptedTerms[key]);
	};

	return (
		<StyledRegisterModal>
			<StyledLabel>
				<StyledCloseButton type="button" aria-label="Close modal" onClick={closeModal}>
					<CloseIcon />
				</StyledCloseButton>
				<SafeHtml html={t('common:registerTitle')} />
			</StyledLabel>
			<StyledModalContent>
				<Form onSubmit={handleSubmit}>
					<InputField
						icon={MdPermContactCalendar}
						name="fullname"
						placeholder={t('common:fullName')}
						type="text"
						validation={{ required: true }}
					/>
					<InputField
						icon={MdMailOutline}
						name="email"
						placeholder="E-mail"
						type="email"
						validation={{ required: true }}
					/>
					<InputField
						icon={MdVpnKey}
						name="password"
						placeholder="Senha"
						type="password"
					/>
					{error ? (
						<ErrorMessage>{message}</ErrorMessage>
					) : (
						<SuccessMessage>{message}</SuccessMessage>
					)}
					<CheckBoxField
						name="terms_conditions"
						value={acceptedTerms.terms_conditions}
						label={
							<Label>
								Concordo com a{' '}
								<Link href="/privacy-policy">
									<a target="_blank" rel="noopener noreferrer">
										Política de Privacidade
									</a>
								</Link>
								e os{' '}
								<Link href="/terms-of-use">
									<a target="_blank" rel="noopener noreferrer">
										Termos e Condições de Uso
									</a>
								</Link>
								.
							</Label>
						}
						required
						onChange={() => handleAcceptedTerms('terms_conditions')}
					/>
					<CheckBoxField
						name="data_conditions"
						value={acceptedTerms.data_conditions}
						label={
							<Label>
								Concordo com o processamento dos meus dados pessoais para fins de
								fornecimento dos serviços da Plataforma Sabiá. Veja mais na{' '}
								<Link href="/privacy-policy">
									<a target="_blank" rel="noopener noreferrer">
										Política de Privacidade
									</a>
								</Link>
								.
							</Label>
						}
						required
						onChange={() => handleAcceptedTerms('data_conditions')}
					/>
					<CheckBoxField
						name="platform_conditions"
						value={acceptedTerms.platform_conditions}
						label={
							<Label>
								Concordo em respeitar a legislação brasileira vigente no conteúdo
								que eu venha a disponibilizar na Plataforma Sabiá, sendo de minha
								exclusiva responsabilidade. Veja mais nos{' '}
								<Link href="/terms-of-use">
									<a target="_blank" rel="noopener noreferrer">
										Termos e Condições de Uso
									</a>
								</Link>
								.
							</Label>
						}
						required
						onChange={() => handleAcceptedTerms('platform_conditions')}
					/>
					<CheckBoxField
						name="services_conditions"
						value={acceptedTerms.services_conditions}
						label={
							<Label>
								Estou ciente de que posso revogar o consentimento de uso dos meus
								dados pessoais a qualquer momento. Todavia, não poderei mais
								utilizar os serviços da plataforma que necessitam do uso e da coleta
								de dados pessoais. Veja mais na{' '}
								<Link href="/privacy-policy">
									<a target="_blank" rel="noopener noreferrer">
										Política de Privacidade
									</a>
								</Link>
								.
							</Label>
						}
						required
						onChange={() => handleAcceptedTerms('services_conditions')}
					/>
					<CheckBoxField
						name="channel_conditions"
						value={acceptedTerms.channel_conditions}
						label={
							<Label>
								Estou ciente quanto ao canal de suporte da Plataforma Sabiá, que
								estará à disposição para sanar eventual dúvida que possa surgir.
							</Label>
						}
						required
						onChange={() => handleAcceptedTerms('channel_conditions')}
					/>
					<CheckBoxField
						name="link_conditions"
						value={acceptedTerms.link_conditions}
						label={
							<Label>
								Estou ciente que em hipótese alguma será constituído vínculo
								cooperativo, associativo, societário ou empregatício entre a
								plataforma, os usuários cadastrados e os parceiros. Veja mais nos{' '}
								<Link href="/terms-of-use">
									<a target="_blank" rel="noopener noreferrer">
										Termos e Condições de Uso
									</a>
								</Link>
								.
							</Label>
						}
						required
						onChange={() => handleAcceptedTerms('link_conditions')}
					/>
					<CheckBoxField
						name="process_conditions"
						value={acceptedTerms.process_conditions}
						label={
							<Label>
								Concordo com o processamento dos meus dados pessoais com o objetivo
								de receber publicidade da Plataforma Sabiá e de terceiros parceiros.
								Veja mais nos{' '}
								<Link href="/terms-of-use">
									<a target="_blank" rel="noopener noreferrer">
										Termos e Condições de Uso
									</a>
								</Link>
								.(Opcional)
							</Label>
						}
						onChange={() => handleAcceptedTerms('process_conditions')}
					/>

					<ActionsRegister>
						<Button type="submit" disabled={loading || !isMissingCheck()}>
							{loading ? t('common:wait') : t('common:register')}
						</Button>
						<LabelGroups>
							<StyledSpan>{t('common:alreadyHaveAnAccount?')}</StyledSpan>
							<StyledLink onClick={() => openModal('login')}>
								{t('common:enterHere')}
							</StyledLink>
						</LabelGroups>
					</ActionsRegister>
				</Form>
			</StyledModalContent>
		</StyledRegisterModal>
	);
};

RegisterModal.propTypes = {
	closeModal: PropTypes.func.isRequired,
};

export default RegisterModal;
