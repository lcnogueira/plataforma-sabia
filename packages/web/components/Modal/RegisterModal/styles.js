import styled, { css } from 'styled-components';
import { Modal } from '../styles';

export const StyledRegisterModal = styled(Modal)`
	padding: 0rem;
	max-width: 67rem;
	width: 100%;
	overflow-y: scroll;
	overflow-x: hidden;

	@media (max-width: ${({ theme }) => theme.screens.medium}px) {
		width: 90%;
		padding: 0;
	}
`;

export const StyledLabel = styled.div`
	padding: 4rem;
	font-family: 'Museo', sans-serif;
	font-weight: 500;
	color: ${({ theme }) => theme.colors.white};
	margin-bottom: 4rem;
	font-size: 3rem;
	background-color: ${({ theme }) => theme.colors.primary};

	> span:last-child {
		word-break: break-word;
	}
`;

export const StyledCloseButton = styled.button`
	${({ theme: { colors, sizes } }) => css`
		background: 0;
		border: 0;
		position: absolute;
		top: 1rem;
		right: 1rem;

		svg {
			height: ${sizes.defaultIcon}rem;
			width: ${sizes.defaultIcon}rem;
			transition: color 0.3s;
			color: ${colors.lightWhite};

			:hover,
			:focus {
				color: ${colors.black};
			}
		}
	`}
`;

export const StyledModalContent = styled.div`
	${({ theme: { screens } }) => css`
		padding-left: 5.8rem;
		padding-right: 5.8rem;

		@media (max-width: ${screens.medium}px) {
			padding-left: 2rem;
			padding-right: 2rem;
		}
	`}
`;

export const LabelGroups = styled.div`
	width: 50%;
	font-size: 1.5rem;
	font-weight: 100;
	padding-left: 2rem;
	display: flex;
	flex-direction: row;
`;

export const Label = styled.label``;

export const ActionsRegister = styled.div`
	${({ theme: { colors, screens } }) => css`
		width: 100%;
		margin-top: 1rem;
		display: flex;
		flex-direction: row;
		align-items: center;
		font: sans-serif;
		button {
			background-color: ${colors.secondary};
			padding: 1rem;
			font: 1em;
			width: 50%;
			font-weight: 200;
		}

		@media screen and (max-width: ${screens.small}px) {
			flex-wrap: wrap;

			button {
				flex-basis: 100%;
				margin-bottom: 1.2rem;
			}

			${LabelGroups} {
				width: 100%;
				padding-left: 0;
			}
		}
	`}
`;
export const StyledSpan = styled.div`
	color: ${({ theme }) => theme.colors.lightGray};
	font-weight: 500;
	margin: 0rem;
	padding-right: 0.5rem;
`;
export const StyledLink = styled.a`
	color: ${({ theme }) => theme.colors.primary};
	font-weight: 500;
	margin: 0rem !important;
	padding: 0rem !important;
`;
