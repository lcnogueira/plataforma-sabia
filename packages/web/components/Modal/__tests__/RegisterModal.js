import React from 'react';

import { render, screen } from 'test-utils';
import RegisterModal from '../RegisterModal';

const closeModalMock = jest.fn();

describe('<RegisterModal />', () => {
	it('should render correctly', () => {
		const { container } = render(<RegisterModal closeModal={closeModalMock} />);

		expect(screen.getByText('common:registerTitle')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});
