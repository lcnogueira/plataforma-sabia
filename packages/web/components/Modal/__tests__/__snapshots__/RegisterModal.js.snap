// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterModal /> should render correctly 1`] = `
.c0 {
  position: fixed;
  min-width: 40rem;
  max-height: 100vh;
  z-index: 2000;
  padding: 4rem;
  background: hsla(0,0%,100%);
  border-radius: 0.5rem;
  box-shadow: 0px 0px 20px -5px rgba(38,38,38,1);
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c8 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  opacity: 1;
  background: hsla(0,0%,100%);
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c8::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c8::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c8:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c8::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c6 {
  font-size: 1.4rem;
  line-height: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem;
  font: 1.2rem;
}

.c10 {
  margin: 0rem;
  width: 1px;
  position: relative;
  left: 2.5rem;
  z-index: -1;
}

.c10:checked + span {
  background: hsla(204,70%,53%) -19px top no-repeat;
  border-color: hsla(204,70%,53%);
}

.c10:checked + span:before {
  content: '';
  position: absolute;
  top: 16%;
  right: 34%;
  display: inline-block;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  height: 12px;
  width: 7px;
  border-bottom: 3px solid hsla(0,0%,100%);
  border-right: 3px solid hsla(0,0%,100%);
}

.c12 {
  width: 100%;
  color: hsla(0,0%,40%);
  cursor: pointer;
}

.c12 a {
  color: hsl(166,100%,34%);
  padding: 0 !important;
  -webkit-transition: color 0.2s ease-in-out;
  transition: color 0.2s ease-in-out;
}

.c12 a:hover {
  color: hsla(146,64%,36%);
}

.c11 {
  display: inline-block;
  position: relative;
  top: -1px;
  width: 2.5rem;
  height: 2.5rem;
  margin: -1px 0px 0 0;
  margin-right: 8px;
  vertical-align: middle;
  background: hsla(0,0%,100%);
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  cursor: pointer;
}

.c5 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c5 button,
.c5 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c5 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

.c14 {
  background-color: hsla(29,94%,62%);
  color: hsla(0,0%,100%);
  border-radius: 0.5rem;
  border: none;
  font-size: 2.2rem;
  text-transform: uppercase;
  padding: 1.8rem 6rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  opacity: 0.8;
  cursor: not-allowed;
}

.c14:hover {
  opacity: 0.8;
}

.c1 {
  padding: 0rem;
  max-width: 67rem;
  width: 100%;
  overflow-y: scroll;
  overflow-x: hidden;
}

.c2 {
  padding: 4rem;
  font-weight: 500;
  color: hsla(0,0%,100%);
  margin-bottom: 4rem;
  font-size: 2.5rem;
  background-color: hsla(29,94%,62%);
}

.c2 > span:last-child {
  word-break: break-word;
}

.c3 {
  background: 0;
  border: 0;
  position: absolute;
  top: 1rem;
  right: 1rem;
}

.c3 svg {
  height: 3rem;
  width: 3rem;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
  color: hsla(0,0%,100%,0.7);
}

.c3 svg:hover,
.c3 svg:focus {
  color: hsla(204,7%,15%);
}

.c4 {
  padding-left: 5.8rem;
  padding-right: 5.8rem;
}

.c16 {
  width: 50%;
  font-size: 1.5rem;
  font-weight: 100;
  padding-left: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c13 {
  width: 100%;
  margin-top: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font: sans-serif;
}

.c13 button {
  background-color: hsl(166,100%,34%);
  padding: 1rem;
  font: 1em;
  width: 50%;
  font-weight: 200;
}

.c17 {
  color: hsla(0,0%,40%);
  font-weight: 500;
  margin: 0rem;
  padding-right: 0.5rem;
}

.c18 {
  color: hsla(29,94%,62%);
  font-weight: 500;
  margin: 0rem !important;
  padding: 0rem !important;
}

@media (max-width:768px) {
  .c0 {
    min-width: 90%;
    padding: 2rem;
    margin: 1rem;
    overflow-x: hidden;
    overflow-y: scroll;
  }
}

@media (max-width:768px) {
  .c14 {
    font-size: 1.6rem;
    padding: 1.4rem 6rem;
  }
}

@media (max-width:768px) {
  .c1 {
    width: 90%;
    padding: 0;
  }
}

@media (max-width:768px) {
  .c4 {
    padding-left: 2rem;
    padding-right: 2rem;
  }
}

@media screen and (max-width:450px) {
  .c13 {
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }

  .c13 button {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    margin-bottom: 1.2rem;
  }

  .c13 .c15 {
    width: 100%;
    padding-left: 0;
  }
}

<div>
  <div
    class="c0 c1"
  >
    <div
      class="c2"
    >
      <button
        aria-label="Close modal"
        class="c3"
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 0 1-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
          />
        </svg>
      </button>
      common:registerTitle
    </div>
    <div
      class="c4"
    >
      <form
        class="c5"
      >
        <div
          class="c6"
        >
          
          <div
            class="c7"
          >
            <input
              aria-label=""
              aria-required="true"
              class="c8"
              id="fullname"
              name="fullname"
              placeholder="common:fullName *"
              type="text"
            />
          </div>
        </div>
        <div
          class="c6"
        >
          
          <div
            class="c7"
          >
            <input
              aria-label=""
              aria-required="true"
              class="c8"
              id="email"
              name="email"
              placeholder="E-mail *"
              type="email"
            />
          </div>
        </div>
        <div
          class="c6"
        >
          
          <div
            class="c7"
          >
            <input
              aria-label=""
              class="c8"
              id="password"
              name="password"
              placeholder="Senha"
              type="password"
            />
          </div>
        </div>
        <p />
        <div
          class="c9"
        >
          <input
            aria-label="[object Object]"
            aria-required="true"
            class="c10"
            form="[object Object]"
            id="terms_conditions"
            name="terms_conditions"
            required=""
            type="checkbox"
          />
          <span
            aria-checked="false"
            aria-label="[object Object]"
            aria-required="true"
            class="c11"
            role="checkbox"
          />
          <label
            class="c12"
            for="terms_conditions"
          >
            <p
              class="styles__Label-l1pr4p-5"
            >
              Concordo com a
               
              <a
                rel="noopener noreferrer"
                target="_blank"
              >
                Política de Privacidade
              </a>
              e os
               
              <a
                rel="noopener noreferrer"
                target="_blank"
              >
                Termos e Condições de Uso
              </a>
              .
            </p>
          </label>
        </div>
        <div
          class="c9"
        >
          <input
            aria-label="[object Object]"
            aria-required="true"
            class="c10"
            form="[object Object]"
            id="data_conditions"
            name="data_conditions"
            required=""
            type="checkbox"
          />
          <span
            aria-checked="false"
            aria-label="[object Object]"
            aria-required="true"
            class="c11"
            role="checkbox"
          />
          <label
            class="c12"
            for="data_conditions"
          >
            <p
              class="styles__Label-l1pr4p-5"
            >
              Concordo com o processamento dos meus dados pessoais para fins de fornecimento dos serviços da Plataforma Sabiá. Veja mais na
               
              <a
                rel="noopener noreferrer"
                target="_blank"
              >
                Política de Privacidade
              </a>
              .
            </p>
          </label>
        </div>
        <div
          class="c9"
        >
          <input
            aria-label="[object Object]"
            aria-required="true"
            class="c10"
            form="[object Object]"
            id="platform_conditions"
            name="platform_conditions"
            required=""
            type="checkbox"
          />
          <span
            aria-checked="false"
            aria-label="[object Object]"
            aria-required="true"
            class="c11"
            role="checkbox"
          />
          <label
            class="c12"
            for="platform_conditions"
          >
            <p
              class="styles__Label-l1pr4p-5"
            >
              Concordo em respeitar a legislação brasileira vigente no conteúdo que eu venha a disponibilizar na Plataforma Sabiá, sendo de minha exclusiva responsabilidade. Veja mais nos
               
              <a
                rel="noopener noreferrer"
                target="_blank"
              >
                Termos e Condições de Uso
              </a>
              .
            </p>
          </label>
        </div>
        <div
          class="c9"
        >
          <input
            aria-label="[object Object]"
            aria-required="true"
            class="c10"
            form="[object Object]"
            id="services_conditions"
            name="services_conditions"
            required=""
            type="checkbox"
          />
          <span
            aria-checked="false"
            aria-label="[object Object]"
            aria-required="true"
            class="c11"
            role="checkbox"
          />
          <label
            class="c12"
            for="services_conditions"
          >
            <p
              class="styles__Label-l1pr4p-5"
            >
              Estou ciente de que posso revogar o consentimento de uso dos meus dados pessoais a qualquer momento. Todavia, não poderei mais utilizar os serviços da plataforma que necessitam do uso e da coleta de dados pessoais. Veja mais na
               
              <a
                rel="noopener noreferrer"
                target="_blank"
              >
                Política de Privacidade
              </a>
              .
            </p>
          </label>
        </div>
        <div
          class="c9"
        >
          <input
            aria-label="[object Object]"
            aria-required="true"
            class="c10"
            form="[object Object]"
            id="channel_conditions"
            name="channel_conditions"
            required=""
            type="checkbox"
          />
          <span
            aria-checked="false"
            aria-label="[object Object]"
            aria-required="true"
            class="c11"
            role="checkbox"
          />
          <label
            class="c12"
            for="channel_conditions"
          >
            <p
              class="styles__Label-l1pr4p-5"
            >
              Estou ciente quanto ao canal de suporte da Plataforma Sabiá, que estará à disposição para sanar eventual dúvida que possa surgir.
            </p>
          </label>
        </div>
        <div
          class="c9"
        >
          <input
            aria-label="[object Object]"
            aria-required="true"
            class="c10"
            form="[object Object]"
            id="link_conditions"
            name="link_conditions"
            required=""
            type="checkbox"
          />
          <span
            aria-checked="false"
            aria-label="[object Object]"
            aria-required="true"
            class="c11"
            role="checkbox"
          />
          <label
            class="c12"
            for="link_conditions"
          >
            <p
              class="styles__Label-l1pr4p-5"
            >
              Estou ciente que em hipótese alguma será constituído vínculo cooperativo, associativo, societário ou empregatício entre a plataforma, os usuários cadastrados e os parceiros. Veja mais nos
               
              <a
                rel="noopener noreferrer"
                target="_blank"
              >
                Termos e Condições de Uso
              </a>
              .
            </p>
          </label>
        </div>
        <div
          class="c9"
        >
          <input
            aria-label="[object Object]"
            aria-required="false"
            class="c10"
            form="[object Object]"
            id="process_conditions"
            name="process_conditions"
            type="checkbox"
          />
          <span
            aria-checked="false"
            aria-label="[object Object]"
            aria-required="false"
            class="c11"
            role="checkbox"
          />
          <label
            class="c12"
            for="process_conditions"
          >
            <p
              class="styles__Label-l1pr4p-5"
            >
              Concordo com o processamento dos meus dados pessoais com o objetivo de receber publicidade da Plataforma Sabiá e de terceiros parceiros. Veja mais nos
               
              <a
                rel="noopener noreferrer"
                target="_blank"
              >
                Termos e Condições de Uso
              </a>
              .(Opcional)
            </p>
          </label>
        </div>
        <div
          class="c13"
        >
          <button
            class="c14"
            color="hsla(0, 0%, 100%)"
            disabled=""
            name=""
            title=""
            type="submit"
          >
            common:register
          </button>
          <div
            class="c15 c16"
          >
            <div
              class="c17"
            >
              common:alreadyHaveAnAccount?
            </div>
            <a
              class="c18"
            >
              common:enterHere
            </a>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
